ssl/quic/quic_channel.c

```diff
@@ -1709,9 +1717,15 @@ static int ch_generate_transport_params(QUIC_CHANNEL *ch)
     int wpkt_valid = 0;
     size_t buf_len = 0;
 
-    if (ch->local_transport_params != NULL || ch->got_local_transport_params)
+    if (ch->state != QUIC_CHANNEL_STATE_IDLE &&
+        (ch->local_transport_params != NULL || ch->got_local_transport_params))
         goto err;
 
+    /* Free prior params */
+    OPENSSL_free(ch->local_transport_params);
+    ch->local_transport_params = NULL;
+    ch->got_local_transport_params = 0;
```

Comment:
Under what circumstances will we already have prior params? Aren't we only supposed to do this once per connection?

Reply:
I think the best answer there is that, generally speaking we shouldn't need to call this twice.  The defaults are setup in ch_init, which is called very early.  There are getters and setters for each field, but they appear unused outside of the quic_multistream test
There are getters and setters for all the fields, but they generally only appear to be used in the multistream test

So I suppose its possible that the deferral of the transport param generation was done to avoid the problems in the quic multistream test.  Really unsure though

Reply:
put another way, the parameters are/arguably should be changeable up to the point they are transmitted, though it seems like they _should_ only need to be set once.